### PR TITLE
add create_box in tensor-based module

### PR DESCRIPTION
**Pull request by ZhengyuDiao**
This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d/5263)

### CODE REVIEW

File: cpp/open3d/t/geometry/TriangleMeshFactory.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    // Mesh.
+    TriangleMesh mesh(vertex_positions, triangle_indices);
+
+    return mesh;
+}
+
+}  // namespace geometry
+}  // namespace t
+}  // namespace open3d
```
Comment (reyanshsolis) [inline cpp/open3d/t/geometry/TriangleMeshFactory.cpp:100]: always leave a newline at the end of the code. There is also some setting to do that in vs-code. https://stackoverflow.com/a/44704969

**Review by reyanshsolis — COMMENTED** [review]
Are you planning to add the uv_map attribute as well?

**Review by yxlao — APPROVED** [review]

**Comment by yxlao** [comment]
LGTM after CI.

**Comment by yxlao** [comment]
> Are you planning to add the uv_map attribute as well?

Have we designed the data structure for UV map for t-mesh? Do we have an example for that?

For this PR, it looks good, and we can add UV map support to this later.

File: cpp/open3d/t/geometry/TriangleMeshFactory.cpp (hunk 1 above)
Comment (ZhengyuDiao) [reply cpp/open3d/t/geometry/TriangleMeshFactory.cpp]: ok, got it

**Pull request merged**